nautobot/extras/api/serializers.py

```diff
@@ -343,6 +344,18 @@ class ConfigContextSerializer(ValidatedModelSerializer, NotesSerializerMixin):
     )
     tags = serializers.SlugRelatedField(queryset=Tag.objects.all(), slug_field="slug", required=False, many=True)
 
+    # Conditional enablement of dynamic groups filtering
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+        if settings.CONFIG_CONTEXT_DYNAMIC_GROUPS_ENABLED:
+            self.fields["dynamic_groups"] = SerializedPKRelatedField(
+                queryset=DynamicGroup.objects.all(),
+                serializer=NestedDynamicGroupSerializer,
+                required=False,
+                many=True,
+            )
+
```

Comment:
Does the `Meta.fields` need to be updated in the `__init__` ?

Reply:
From my understanding of the documentation I don't believe so.

Reply:
As long as the field is defined on the class it doesn't need to be in `Meta.fields`

Reply:
Okay, I'll remove it then.

Reply:
OH wait, this may not be true for serializers. My apologies. I was thinking of filtersets.

Reply:
Yeah had to add it back, tests started complaining.